lib/Monads/NonDetMonadVCG.thy

```diff
@@ -709,6 +711,11 @@ lemma hoare_vcg_if_split:
   "\<lbrakk> P \<Longrightarrow> \<lbrace>Q\<rbrace> f \<lbrace>S\<rbrace>; \<not>P \<Longrightarrow> \<lbrace>R\<rbrace> g \<lbrace>S\<rbrace> \<rbrakk> \<Longrightarrow> \<lbrace>\<lambda>s. (P \<longrightarrow> Q s) \<and> (\<not>P \<longrightarrow> R s)\<rbrace> if P then f else g \<lbrace>S\<rbrace>"
   by simp
 
+lemma hoare_vcg_if_split_strong:
+ "\<lbrakk> P \<Longrightarrow> \<lbrace>Q\<rbrace> f \<lbrace>S\<rbrace>; \<not>P \<Longrightarrow> \<lbrace>R\<rbrace> g \<lbrace>S\<rbrace> \<rbrakk> \<Longrightarrow>
+  \<lbrace>\<lambda>s. if P then Q s else R s\<rbrace> if P then f else g \<lbrace>S\<rbrace>"
```

Comment:
I wouldn't call that rule `_strong`, it's not stronger than the existing rules and it doesn't really fit and automatic wp or sp calculus (for sp it should be something like an `S and S'`, because it is unlikely that both branches will have the same calculated post condition). How about `hoare_vcg_if_post_join`?

Reply:
It looks like this isn't used in rt anymore, the lemma it was written for was redone at some point. Is it worth keeping or should I just remove it?

Reply:
I'd remove it